torchvision/datasets/lsun.py

```diff
@@ -93,6 +94,16 @@ def __init__(
             self.indices.append(count)
 
         self.length = count
+        
+    def _download(self, c):
+        url = 'http://dl.yf.io/lsun/scenes/{}'.format(c + '_lmdb.zip')
```

Comment:
> Do I have to find md5 hash [...] otherwise I can just directly check if the file exists or not

Although not terribly common, download errors happen. Without a checksum, you have no idea if the file maybe is corrupt. Thus it we prefer to have a MD5 checksum for most files. In your case that would mean for the `.zip` and `.lmdb` files of every class.

> if yes how?

You can download and extract all archives with the download logic you already have. Afterwards you can use a variety of tools. Depending on your setup you might have [`md5sum`](https://wiki.ubuntuusers.de/md5sum/) already installed. If you don't want additional software, you can use `torchvision.datasets.utils.calculate_md5`

https://github.com/pytorch/vision/blob/de908627640151defa27be0fdd741720a0940048/torchvision/datasets/utils.py#L26